File: Classes/Graphics and Tools/Text/DrawFormEntry.m

#import "DrawFormEntry.h"

#import "DrawFunctions.h"

#import <AJRInterfaceFoundation/AJRInterfaceFoundation.h>

@implementation DrawFormEntry

- (void)setupTextView:(NSTextView *)textView {
    [super setupTextView:textView];

    [textView setRichText:NO];
    [textView setImportsGraphics:NO];
}

- (DrawGraphicCompletionBlock)drawPath:(AJRBezierPath *)path withPriority:(DrawAspectPriority)priority {
    DrawGraphicCompletionBlock completionBlock = [super drawPath:path withPriority:priority];

    [[NSColor lightGrayColor] set];
    [path stroke];

    return completionBlock;
}

@end

File: Classes/Graphics and Tools/Text/DrawFormEntry.swift
/*
DrawFormEntry.m
Draw

Copyright © 2021, AJ Raftis and AJRFoundation authors
All rights reserved.

Redistribution and use in source and binary forms, with or without modification,
are permitted provided that the following conditions are met:

* Redistributions of source code must retain the above copyright notice, this 
  list of conditions and the following disclaimer.
* Redistributions in binary form must reproduce the above copyright notice, 
  this list of conditions and the following disclaimer in the documentation 
  and/or other materials provided with the distribution.
* Neither the name of Draw nor the names of its contributors may be 
  used to endorse or promote products derived from this software without 
  specific prior written permission.

THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED 
WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE 
DISCLAIMED. IN NO EVENT SHALL AJ RAFTIS BE LIABLE FOR ANY DIRECT, INDIRECT, 
INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR 
PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF 
LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF 
ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
*/

import AJRInterface

@objcMembers
open class DrawFormEntry : DrawText {

    open override func setupTextView(_ textView: NSTextView) {
        super.setupTextView(textView)

        textView.isRichText = false
        textView.importsGraphics = false
    }

    // MARK: - DrawAspect

    open override func draw(_ path: AJRBezierPath, with priority: DrawAspectPriority) -> DrawGraphicCompletionBlock? {
        let completionBlock = super.draw(path, with: priority)
        path.stroke(color: NSColor.lightGray)
        return completionBlock
    }

    // MARK: - AJRXMLCoding

    class open override var ajr_nameForXMLArchiving: String {
        return "formEntry"
    }

}